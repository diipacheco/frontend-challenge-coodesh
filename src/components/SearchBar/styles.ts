import styled from 'styled-components';

export const Container = styled.div`
  display: flex;

  width: 100%;
  height: 40px;

  position: relative;
  margin-top: 26px;

  > input {
    width: 100%;
    height: 100%;

    padding: 0 13px;
    border-radius: 4px;
    border: 2px solid var(--grey);

    &::placeholder {
      font-weight: bold;
      color: var(--grey);
      font-size: 15px;
    }
  }

  > svg {
    position: absolute;
    right: 13px;
    top: 12px;
  }
`;
