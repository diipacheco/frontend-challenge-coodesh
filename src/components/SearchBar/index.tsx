import React from 'react';
import { RiUserSearchLine } from 'react-icons/ri';

import { Container } from './styles';

const SearchBar: React.FC = () => {
  return (
    <Container>
      <input type="text" placeholder="Searching" />
      <RiUserSearchLine />
    </Container>
  );
};

export default SearchBar;
