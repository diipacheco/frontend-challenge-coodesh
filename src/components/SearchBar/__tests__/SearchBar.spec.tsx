import { render } from '@testing-library/react';

import SearchBar from '../index';

describe('<SearchBar />', () => {
  it('should match snapshots', () => {
    const wrapper = render(<SearchBar />);

    expect(wrapper).toMatchSnapshot();
  });
});
