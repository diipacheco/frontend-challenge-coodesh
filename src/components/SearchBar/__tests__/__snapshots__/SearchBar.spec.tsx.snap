// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBQB hQNJjl"
      >
        <input
          placeholder="Searching"
          type="text"
        />
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 14v2a6 6 0 0 0-6 6H4a8 8 0 0 1 8-8zm0-1c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm9.446 9.032l1.504 1.504-1.414 1.414-1.504-1.504a4 4 0 1 1 1.414-1.414zM18 20a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
            />
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBQB hQNJjl"
    >
      <input
        placeholder="Searching"
        type="text"
      />
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 14v2a6 6 0 0 0-6 6H4a8 8 0 0 1 8-8zm0-1c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm9.446 9.032l1.504 1.504-1.414 1.414-1.504-1.504a4 4 0 1 1 1.414-1.414zM18 20a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
          />
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
